snuba/query/logical.py

```diff
@@ -59,6 +59,7 @@ def __init__(
         # in order not to expose the internal representation.
         self.__body = body
         self.__final = False
+        self.__sample = sample
```

Comment:
Why do you need final and sample on the Logical query? Shouldn't they be set on the Entity?

Reply:
Ideally yes, but that would be a bigger refactoring to do since now the language allows for sampling a query, so the query object needs a sampling field before we select the entity. We need to be able to propagate the sampling rate from the query body to the entity object. 
Here I just moved it down because it was already in the abstract query class.

Reply:
OK. Can you add a TODO or task or something to document this?